### DISCUSSION TITLE
How to activate appium plugins with Appium beta 2.0?

### DISCUSSION THREAD

**Discussion by Balakumar-14** [appium/appium#17269] [Q&A]
**While trying to start server:**

Welcome to Appium v2.0.0-beta.40 (REV 0b030b74e2ac518bd0bc4158f96c449198f9957f)
[Appium] Non-default server args:
[Appium] {
[Appium]   basePath: '/wd/hub'
[Appium] }
[Appium] Attempting to load driver xcuitest...
[debug] [Appium] Requiring driver at /Users/bala-14246/.appium/node_modules/appium-xcuitest-driver
[Appium] Attempting to load driver uiautomator2...
[debug] [Appium] Requiring driver at /Users/bala-14246/.appium/node_modules/appium-uiautomator2-driver
[Appium] Appium REST http interface listener started on 0.0.0.0:4723
[Appium] Available drivers:
[Appium]   - xcuitest@4.7.4 (automationName 'XCUITest')
[Appium]   - uiautomator2@2.4.5 (automationName 'UiAutomator2')
[Appium] Available plugins:
[Appium]   - images@1.3.2
[Appium]   - execute-driver@2.0.2
[Appium]   - relaxed-caps@1.0.0-beta.9
[Appium] No plugins activated. Use the --use-plugins flag with names of plugins to activate

### So, trying to activate plugin:

appium --plugins=images@1.3.2
WARN Appium Appium encountered 2 warnings while validating plugins found in manifest /Users/bala-14246/.appium/node_modules/.cache/appium/extensions.yaml
WARN Appium Plugin "element-wait" has 1 potential problem: 
WARN Appium   - Plugin "element-wait" (package appium-wait-plugin) may be incompatible with the current version of Appium (v2.0.0-beta.40) due to an invalid or missing peer dependency on Appium. Please ask the developer of appium-wait-plugin to add a peer dependency on ^appium@2.0.0-beta.40.
usage: appium [-h] [-v] {server,driver,plugin} ...

_**appium: error: unrecognized arguments: --plugins=images@1.3.2**_

So, could anyone help me with activating the plugins?

**Comment by saikrishna321** [comment]
Try --use-plugins

**Comment by Balakumar-14** [comment]
Thank you, @saikrishna321.
Can we enable plugins permanently?

**Reply by saikrishna321** [reply]
You need to call this on every server start

**Reply by Balakumar-14** [reply]
[debug] [Appium] Requiring driver at /Users/bala-14246/.appium/node_modules/appium-xcuitest-driver 
Does this cause any issues in the future?

**Reply by saikrishna321** [reply]
Sorry, can you be more specific?

**Reply by jlipps** [reply]
this log line is normal

**Reply by Balakumar-14** [reply]
Thank you both @jlipps @saikrishna321 !

**I am getting this error after installing beta 2.0. Could you please help me with this? As I am a beginner.**

[debug] [HTTP] No route found for /wd/hub/status
[HTTP] <-- GET /wd/hub/status 404 1 ms - 211

I mentioned my capabilities below:
`{
    "capabilities": 
	[
        {
            "platform": "ANDROID",
            "deviceName": "Pixel_5_Bala",
            "browserName": "android",
            "version": "12",
            "maxInstances": 1        }
    ],
    "configuration":
 {
        "cleanUpCycle": 3000,
        "timeout": 6000,
        "proxy": "org.openqa.grid.selenium.proxy.DefaultRemoteProxy",
        "host": "127.0.0.1",
        "port": 4740,
        "url": "http://127.0.0.1:4740",
        "maxSession": 1,
        "register": true,
        "registerCycle": 5000,
        "hubPort": 4444,
        "hubHost": "172.23.134.87"
    }   
}`

Command to start server: 
`appium --session-override --nodeconfig /Users/bala-14246/IdeaProjects/PORTS/JSON_FILES/Pixel_5_Bala_emulator.json -p 4740`

**Reply by RSM-SBA** [reply]
@Balakumar-14 
i think /wd/hub/ is not necessary anymore in appium 2.0
most probably you have set this at driver initialization

**Reply by Balakumar-14** [reply]
How to solve this issue? @StephanB75
And, i have not used /wd/hub/

**Reply by Balakumar-14** [reply]
@saikrishna321 Could you help me with this?

**Reply by RSM-SBA** [reply]
how you initialize the driver ?
try delete the appium settings from emulator
![image](https://user-images.githubusercontent.com/90029825/181170347-0a3e34bf-c361-40da-a498-62e7761cbc4a.png)

**Reply by Balakumar-14** [reply]
Installed drivers with this reference:
https://www.headspin.io/blog/installing-appium-2-0-and-the-driver-and-plugins-cli

Tried by deleting the appium settings from emulator. But, same error occurs.

**Reply by RSM-SBA** [reply]
i ment not the driver installation.

How you connect the driver to the server ?
something like
driver = new AndroidDriver(new URL("http://127.0.0.1:4740), options);

**Comment by Balakumar-14** [comment]
I haven't initialized still now. I have just started the server and noticed this debug status error. (Just tried to connect node with hub. Node is connected though). Even, I could access the emultor with this server in appium inspector.

**Comment by saikrishna321** [comment]
@Balakumar-14 Start the appium server with args `appium server -ka 800 -pa /wd/hub ` 

`new AndroidDriver(new URL("http://127.0.0.1:4740/wd/hub), options);`
